src/analysis/processing/qgsalgorithmhttprequest.cpp

```diff
@@ -0,0 +1,200 @@
+
+  std::unique_ptr< QgsProcessingParameterEnum > methodParam = std::make_unique < QgsProcessingParameterEnum > (
+        QStringLiteral( "METHOD" ),
+        QObject::tr( "Method" ),
+        QStringList()
+        << QObject::tr( "GET" )
+        << QObject::tr( "POST" ),
+        false,
+        0
+      );
+  methodParam->setHelp( QObject::tr( "The HTTP method to use for the request" ) );
+  addParameter( methodParam.release() );
+
+  std::unique_ptr< QgsProcessingParameterString > dataParam = std::make_unique < QgsProcessingParameterString >(
+        QStringLiteral( "DATA" ), tr( "Data" ), QVariant(), false, true );
+  dataParam->setHelp( QObject::tr( "The data to add in the body if the request is a POST" ) );
+  addParameter( dataParam.release() );
+
+  std::unique_ptr< QgsProcessingParameterBoolean > failureParam = std::make_unique < QgsProcessingParameterBoolean >(
+        QStringLiteral( "FAILONERROR" ), tr( "Consider error feedback as failure" ), false );
+  failureParam->setHelp( tr( "Let the algorithm fail on error" ) );
+  addParameter( failureParam.release() );
+
+  std::unique_ptr< QgsProcessingParameterBoolean > writeToFileParam = std::make_unique < QgsProcessingParameterBoolean >(
+        QStringLiteral( "WRITETOFILE" ), tr( "Write reply data to temp file" ), false );
+  writeToFileParam->setHelp( tr( "Write the reply data to a temporary binary file instead of passing back in a string" ) );
+  addParameter( writeToFileParam.release() );
+
+  addOutput( new QgsProcessingOutputNumber( QStringLiteral( "STATUSCODE" ), QObject::tr( "HTTP status code" ) ) );
+  addOutput( new QgsProcessingOutputNumber( QStringLiteral( "ERRORCODE" ), QObject::tr( "Network error code" ) ) );
+  addOutput( new QgsProcessingOutputString( QStringLiteral( "ERRORMESSAGE" ), QObject::tr( "Network error message" ) ) );
+  addOutput( new QgsProcessingOutputString( QStringLiteral( "RESULTDATA" ), QObject::tr( "Reply data or path to temporary file keeping the reply data" ) ) );
+}
+
+QVariantMap QgsHttpRequestAlgorithm::processAlgorithm( const QVariantMap &parameters, QgsProcessingContext &context, QgsProcessingFeedback *feedback )
+{
+  const QString url = parameterAsString( parameters, QStringLiteral( "URL" ), context );
+  if ( url.isEmpty() )
+    throw QgsProcessingException( tr( "No URL specified" ) );
+  const Qgis::HttpMethod httpMethod = static_cast< Qgis::HttpMethod>( parameterAsEnum( parameters, QStringLiteral( "METHOD" ), context ) );
+  const QString data = parameterAsString( parameters, QStringLiteral( "DATA" ), context );
+  const bool failOnError  = parameterAsBool( parameters, QStringLiteral( "FAILONERROR" ), context );
+  const bool writeToFile = parameterAsBool( parameters, QStringLiteral( "WRITETOFILE" ), context );
+  const QUrl qurl = QUrl::fromUserInput( url );
+
+  // Make Request
+  const QNetworkRequest request( qurl );
+  QNetworkReply *reply = nullptr;
+
+  QEventLoop loop;
```

Comment:
Instead of a local event loop, use QgsBlockingNetworkRequest instead. That's safer to use vs the issues which come with local event loops.